### DISCUSSION TITLE
Give the option in the endpoint section to select the microrservice based on http verb or route

### DISCUSSION THREAD

**Discussion by edocrippaofficial** [mia-platform/community#151] [Ideas]
Hi, it happened us a few times to configure the api-gateway in the advanced section to be able to redirect request to  different microservices based on the http verb.

Right now, we have to configure the files: maps-proxyUrl.before.map and maps-proxyName.before.map (or the endpoints.yaml if we are using Envoy).

As an example, i have to redirect my request /users to:
- POST: my bff, to add custom logic
- GET: crud-serivce, to retreive the data

It would be nice to be able to specify different destinations and relative rewrite in the routes section of the endpoint.

What do you think?

@AlexBona 

![image](https://user-images.githubusercontent.com/39826608/209353194-a6221a6b-f60f-4454-8b63-25e5ddfa11c4.png)

**Comment by theliuk** [comment]
I think it's a good idea. @davidebianchi @fredmaggiowski @Danielecina , what do you think?

**Reply by fredmaggiowski** [reply]
I agree that this could be a nice feature to have; it should be best to track it in the internal backlog @albertotessarotto @jacoposte FYI

**Reply by jacoposte** [reply]
Nice idea, 
we track it and we plan it when it's possibile.
Thanks!

**Reply by edocrippaofficial** [reply]
Hi @fredmaggiowski, @jacoposte 
Is there any news on this feature?

**Reply by theliuk** [reply]
This feature is very nice, and it allows the use to fully exploit the concept of api-gateway. We find it very useful for backoffice projects, where information gathering is typically performed by a CRUD endpoint, but the creation of entities is exposed by some other microservice.

**Reply by fredmaggiowski** [reply]
As far as I know there's currently no plan for this feature

**Reply by theliuk** [reply]
But is there a way to participate to the decision of wether something is in roadmap or not?

**Reply by fredmaggiowski** [reply]
follow the roadmap and vote the issue

in the meantime @albertotessarotto @jacoposte @davidebianchi FYI

**Reply by theliuk** [reply]
We're voting hard
